gensim/models/phrases.py

```diff
@@ -105,6 +109,44 @@ class Phrases(interfaces.TransformationABC):
     and `phrases[corpus]` syntax.
 
     """
+    #from gensim.models.phrases_inner import learn_vocab
+    try:
+        from gensim.models.phrases_inner import learn_vocab
+        logger.info("Cython file loaded")
+    except ImportError:
+        logger.info("Cython file not loaded")
+        #failed... fall back to plain numpy (20-80x slower training than the above)
+
+    
+        def learn_vocab(self,sentences, max_vocab_size, delimiter=b'_', progress_per=10000):
+            #Collect unigram/bigram counts from the `sentences` iterable.
+            sentence_no = -1
+            total_words = 0
+            logger.info("collecting all words and their counts")
+            vocab = defaultdict(int)
+            min_reduce = 1
+            for sentence_no, sentence in enumerate(sentences):
+                if sentence_no % progress_per == 0:
+                    logger.info("PROGRESS: at sentence #%i, processed %i words and %i word types" %
+                                (sentence_no, total_words, len(vocab)))
+                #sentence = [utils.any2utf8(w) for w in sentence] 
```

Comment:
Yes, saving memory.

Up to Python 3.3 (and including all Python 2.x), unicode strings take up 2-4x as much memory, compared to UTF8 byte strings.

Since memory is more critical than speed here, we went with UTF8.